client/alloc_runner.go

```diff
@@ -744,6 +754,9 @@ func (r *AllocRunner) Run() {
 	defer close(r.waitCh)
 	go r.dirtySyncState()
 
+	// Increment alloc runner start counter. Incr'd even when restoring existing tasks so 1 start != 1 task execution
+	metrics.IncrCounter([]string{"client", "allocs", r.alloc.Job.Name, r.alloc.TaskGroup, "start"}, 1)
```

Comment:
I would move this below the terminal status check. We don't want to emit it if the alloc_runner is just waiting for a destroy.